pystac/catalog.py

```diff
@@ -730,13 +737,21 @@ def fn() -> None:
             return fn
 
         def process_catalog(
-            cat: Catalog, _root_href: str, is_root: bool
+            cat: Catalog,
+            _root_href: str,
+            is_root: bool,
+            parent: Optional[Catalog] = None,
         ) -> List[Callable[[], None]]:
             setter_funcs: List[Callable[[], None]] = []
 
             if not skip_unresolved:
                 cat.resolve_links()
 
+            # Abort as the intended parent is not the actual parent
+            # https://github.com/stac-utils/pystac/issues/1116
+            if parent is not None and cat.get_parent() != parent:
```

Comment:
What I'm trying here is to figure out whether the parent link and the actual parent in the catalog are not the same and then skip setting the links to avoid that the wrong self link doesn't override the correct self link.